### PR TITLE
Homework1.5

### PR DESCRIPTION
**Pull request by bmukha**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -11,23 +12,23 @@ const App = () => {
 	return (
```
Comment (KitTheProgrammer) [inline src/App.tsx:14]: If you are using TS you have to add types to almost everything you have.
For components I suggest for you to use `React.FC<PropsTypes>`. for useState you can do it like this: `const [a, b] = useState<type>(...)

For handlers that don't return anything you need to write `const handleAddNewCourseButtonClick = (): void => setAddingMode(true);`

File: src/components/Courses/components/SearchBar/SearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import { SEARCH_BUTTON_TEXT } from '../../../../constants';
+
+import StyledSearchBar from './SearchBar.styles';
+
+interface SearchBarProps extends FlexContainerProps {
+	setSearchText: Dispatch<React.SetStateAction<string>>;
+}
+
+const SearchBar = ({ setSearchText }: SearchBarProps) => {
+	const [query, setQuery] = useState('');
```
Comment (KitTheProgrammer) [inline src/components/Courses/components/SearchBar/SearchBar.tsx]: type for useState

File: src/components/Courses/components/SearchBar/SearchBar.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+	const handleInputChange: ChangeEventHandler<HTMLInputElement> = ({
+		target,
+	}) => {
```
Comment (KitTheProgrammer) [inline src/components/Courses/components/SearchBar/SearchBar.tsx]: you can destruct more: `func = ({ target: { value } }) => {}`

File: src/components/CreateCourse/CreateCourse.tsx [hunk 4]
```diff
@@ -14,26 +16,37 @@ import {
...
-}) => {
+}: CreateCourseProps) => {
 	const [title, setTitle] = useState('');
 	const [description, setDescription] = useState('');
 	const [authorName, setAuthorName] = useState('');
-	const [duration, setDuration] = useState('');
-	const [chosenAuthors, setChosenAuthors] = useState([]);
+	const [duration, setDuration] = useState(0);
+	const [chosenAuthors, setChosenAuthors] = useState<Author[]>([]);
 	const [unusedAuthors, setUnusedAuthors] = useState([...authors]);
```
Comment (KitTheProgrammer) [inline src/components/CreateCourse/CreateCourse.tsx]: types for useState

File: src/components/CreateCourse/CreateCourse.tsx [hunk 5]
```diff
@@ -47,11 +60,21 @@ const CreateCourse = ({
...
 	};
 
-	const handleCancelButtonClick = () => {
+	const handleCancelButtonClick: MouseEventHandler<HTMLButtonElement> = () => {
 		setAddingMode(false);
 	};
 
-	const handleCreateCourseButtonClick = (e) => {
+	const handleDurationKeydownEvent = (e: KeyboardEvent) => {
+		const { key } = e;
```
Comment (KitTheProgrammer) [inline src/components/CreateCourse/CreateCourse.tsx]: function return type

File: src/components/CreateCourse/CreateCourse.tsx [hunk 6]
```diff
@@ -87,131 +110,140 @@ const CreateCourse = ({
 		setAddingMode(false);
 	};
 
-	const addCourseAuthor = (id) => {
+	const addCourseAuthor = (id: string) => {
 		const authorToAdd = authors.find((author) => author.id === id);
```
Comment (KitTheProgrammer) [inline src/components/CreateCourse/CreateCourse.tsx]: function return type

File: src/components/CreateCourse/CreateCourse.tsx [hunk 7]
```diff
@@ -87,131 +110,140 @@ const CreateCourse = ({
...
+		if (authorToAdd) {
+			setChosenAuthors([...chosenAuthors, authorToAdd]);
+		}
 		const filteredAuthors = unusedAuthors.filter((author) => author.id !== id);
 		setUnusedAuthors(filteredAuthors);
 	};
 
-	const deleteCourseAuthor = (id) => {
+	const deleteCourseAuthor = (id: string) => {
 		const authorToDelete = authors.find((author) => author.id === id);
```
Comment (KitTheProgrammer) [inline src/components/CreateCourse/CreateCourse.tsx]: same

File: src/helpers/getAuthorsNamesById.ts [hunk 8]
```diff
@@ -0,0 +1,7 @@
+const getAuthorsNamesById = (ids: Course['authors'], authors: Author[]) => {
+	return ids
```
Comment (KitTheProgrammer) [inline src/helpers/getAuthorsNamesById.ts]: and here

File: src/helpers/pipeDuration.ts [hunk 9]
```diff
@@ -1,4 +1,4 @@
-const pipeDuration = (duration) => {
+const pipeDuration = (duration: Course['duration']) => {
 	const hours = Math.trunc(duration / 60);
```
Comment (KitTheProgrammer) [inline src/helpers/pipeDuration.ts]: and here

**Comment by bmukha** [comment]
@KitTheProgrammer I've addressed all issues, please review.

**Review by KitTheProgrammer — APPROVED** [review]

**Pull request merged**